autode/hessians.py

```diff
@@ -213,19 +200,21 @@ def _proj_matrix(self) -> np.ndarray:
 
         # Construct M^1/2, which as it's diagonal, is just the roots of the
         # diagonal elements
-        masses = np.repeat(
-            [atom.mass for atom in self.atoms], repeats=3, axis=np.newaxis
-        )
-        m_half = np.diag(np.sqrt(masses))
+        masses = np.repeat([atom.mass for atom in self.atoms], repeats=3)
+        m_half = np.sqrt(masses)
 
         for t_i in (t1, t2, t3, t4, t5, t6):
-            t_i[:] = np.dot(m_half, np.array(t_i))
-            t_i /= np.linalg.norm(t_i)
+            t_i *= m_half
 
         # Generate a transform matrix D with the first columns as translation/
         # rotation vectors with the remainder as random orthogonal columns
-        M = np.random.rand(3 * len(self.atoms), 3 * len(self.atoms)) - 0.5
-        M[:, :6] = np.column_stack((t1, t2, t3, t4, t5, t6))
+        M = np.eye(3 * len(self.atoms))
```

Comment:
My suggestion editing from this line:

```
 tr_bas = np.array(tr_vecs).transpose()
U_s, s_v, _ = np.linalg.svd(tr_bas, full_matrices=True)

if self.atoms.are_linear() and s_v[5]/s_v[0] > 1.e-4:
    logger.warning("Molecule detected as linear, but lowest singular "
                             "value from trans. rot. basis is {s_v[5]:.4e}")

return U_s
```

If the rank of matrix is k, then first k columns of SVD will be the orthonormal basis for the column space. Since we started from only the translation and rotation vectors, the first 5 or 6 columns of U will be translation and rotation basis - and the rest will be orthogonal vectors corresponding to vibration (since U always produces an orthonormal basis, as far as i understand!)